### PR TITLE
Order fulfillment

### PR DESCRIPTION
**Pull request by Eternal-Flame085**
- created spec for user story 49
- created route for /orders/:item_id
- Created view page
- Model test for merchant_items
- removed save and open page
- created test for user_story 50
- created model method test
- Created method in item_order model
- edited order show page
- created route and controller method
- Created test for user story 51

### CODE REVIEW

File: spec/features/merchant/order_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+      @tire.reload
+      expect(@tire.inventory).to eq(10)
+    end
+
+    it "I  do not see a 'fulfill' button or link
+    Instead I see a notice next to the item indicating I cannot fulfill this item" do
+    @tire.update(inventory: 1)
+    visit "/merchant/orders/#{@order_1.id}"
+    save_and_open_page
```
Comment (SageOfCode) [inline spec/features/merchant/order_show_spec.rb]: Please delete this

File: spec/features/merchant/order_show_spec.rb [hunk 2]
```diff
@@ -62,5 +62,26 @@
...
+
+      expect(page).to have_content("Item has been fulfilled")
+
+      within id="#item-#{@item_order_2.item_id}" do
+        expect(page).to_not have_button('fulfill')
+        expect(page).to have_content("Item has already been fulfilled")
+      end
+
+      @tire.reload
+      expect(@tire.inventory).to eq(10)
```
Comment (JBelta) [inline spec/features/merchant/order_show_spec.rb:84]: Finally we can reduce our inventory.

File: app/views/merchant/dashboard/order.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+<% @order.merchant_items(current_user.merchant.id).each do |item_order|%>
+<section id="item-<%= item_order.item_id %>">
+  Item name: <%= link_to "#{item_order.item.name}", "/items/#{item_order.item_id}" %><br>
+  <%= image_tag item_order.item.image, height: '50', width: '50' %><br>
+  Price: <%= item_order.price %><br>
+  Quantity: <%= item_order.quantity %><br>
+  <% if item_order.status != 'fulfilled' && item_order.item.inventory >= item_order.quantity %>
+    <%= button_to "fulfill", "/merchant/orders/#{item_order.id}", method: :patch %>
+    <% elsif item_order.item.inventory < item_order.quantity  %>
```
Comment (SageOfCode) [inline app/views/merchant/dashboard/order.html.erb]: indentation is not inline with if statement

**Review by SageOfCode — APPROVED** [review]
Looks ready to merge.

File: spec/features/merchant/order_show_spec.rb [hunk 4]
```diff
@@ -88,7 +88,6 @@
     Instead I see a notice next to the item indicating I cannot fulfill this item" do
     @tire.update(inventory: 1)
     visit "/merchant/orders/#{@order_1.id}"
-    save_and_open_page
       within id="#item-#{@item_order_1.item_id}" do
```
Comment (JBelta) [inline spec/features/merchant/order_show_spec.rb:91]: What's red and smells like blue paint
...
red paint.

**Review by araspaas — APPROVED** [review]

File: spec/features/merchant/order_show_spec.rb (hunk 1 above)
Comment (JBelta) [reply spec/features/merchant/order_show_spec.rb]: ... now

**Review by JBelta — APPROVED** [review]

**Pull request merged**